app/services/preserved_object_handler.rb

```diff
@@ -163,25 +195,29 @@ def create_db_objects(status, validated=false)
     results.concat(create_results)
   end
 
-  def update_online_version
+  # TODO: this is "too complex" per rubocop: shameless green implementation
+  # NOTE: if reduce complexity, remove Metrics/PerceivedComplexity exception in .rubocop.yml
+  def update_online_version(validated=false, status=nil)
     results = []
     upd_results = with_active_record_transaction_and_rescue do
       pres_object = PreservedObject.find_by!(druid: druid)
-      pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint)
+      pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
       # FIXME: what if there is more than one associated pres_copy?
       if incoming_version > pres_copy.version && pres_copy.version == pres_object.current_version
         results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_copy.class.name)
-        update_preserved_copy(pres_copy, incoming_version, incoming_size)
-        results.concat(update_status(pres_copy, Status.ok))
+        update_preserved_copy_version_etc(pres_copy, incoming_version, incoming_size, validated)
```

Comment:
what does _etc stand for?

Reply:
"et cetera" because I couldn't think of a better name for the method.  Open to suggestions.

Reply:
updated_pc_obj
updated_pc_version
updated_pc_db_obj

i dunno just throwing names out there...

Reply:
I was trying to say "this is updating a pc object version, along with some other fields"